src/emm/services/modules_service.py

```diff
@@ -225,9 +232,123 @@ def git_commit_modules(self, commit: str) -> None:
+                "{} pr: {}".format(repo, link) for repo, link in comments.items() if repo != module
+            )
+            pr_dict[module] = []
+            pr_dict[module].append(pr_url)
+            pr_dict[module].append(pr_links)
+        return pr_dict
+
+    def update_pr_links(self, comment_links: Dict[str, List[str]]) -> List[str]:
+        """
+        Update link to each module.
+
+        :param comment_links: Dictionary of module with its location and other modules' pull request link.
+        :return: List of pull requests being created associate with its link.
+        """
+        all_pull_requests = []
+        lines = (
+            "This code review is spread across multiple repositories. "
+            "Links to the other components of this review can be found:\n "
+        )
+        for module, pr_list in comment_links.items():
+            pr_url = pr_list[0]
+            if module != "base":
+                module_data = self.get_module_data(module)
+                module_location = Path(module_data.prefix) / module
+                pr_links = lines + pr_list[1]
+                self.github_service.pr_comment(pr_url, pr_links, module_location)
+                all_pull_requests.append(pr_url)
+                all_pull_requests.append(pr_links)
+            else:
+                pr_links = lines + pr_list[1]
+                self.github_service.pr_comment(pr_url, pr_links)
+                all_pull_requests.append(pr_url)
+                all_pull_requests.append(pr_links)
+
+        return all_pull_requests
+
+    def push_branch_to_remote(self, directory: Optional[Path] = None) -> None:
+        """
+        Push current branch to remote if current branch is not protected branch.
+
+        :param directory: Directory to execute the command at.
+        """
+        branch = self.git_service.current_branch(directory)
+        if branch in PROTECTED_BRANCHES:
+            raise ValueError("Cannot push unreviewed modification to master.")
+
+        if not self.git_service.current_branch_exist_on_remote(branch, directory):
+            self.git_service.push_branch_to_remote(directory)
+
+    def module_pull_request(self, args: List[str]) -> Dict[str, str]:
+        """
+        Create pull request for each module.
+
+        :param args: Arguments to pass to the github CLi.
+        :return: Dictionary of module and its pull request url.
+        """
+        comment_dict = {}
+        enabled_modules = self.get_all_modules(True)
+        for module in enabled_modules:
+            module_data = self.get_module_data(module)
+            module_location = Path(module_data.prefix) / module
+            basename = self.git_service.get_base_branch_name(module_location)
+            if basename and self.git_service.check_changes(basename, module_location):
+                self.push_branch_to_remote(module_location)
+            else:
+                raise ValueError("No changes found in current branch.")
```

Comment:
Why raise an exception if there aren't changes? That seems like a valid use-case.